src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTest.cs

```diff
@@ -23,16 +23,33 @@ public SdkTemplateTest(string scenarioName, DotNetLanguage language, string targ
         TargetRid = targetRid;
     }
 
-    internal void Execute(DotNetSdkHelper dotNetHelper, string testRoot, string[]? frameworks = null)
+    internal void Execute(DotNetSdkHelper dotNetHelper, string testRoot, string[]? frameworks = null, string? PreMadeSolution = null)
     {
         // Don't use the cli language name in the project name because it may contain '#': https://github.com/dotnet/roslyn/issues/51692
         string projectName = $"{ScenarioName}_{Template}_{Language}";
         string customNewArgs = Template.IsAspNetCore() && NoHttps ? "--no-https" : string.Empty;
         string projectDirectory = Path.Combine(testRoot, projectName);
 
-        Directory.CreateDirectory(projectDirectory);
-
-        dotNetHelper.ExecuteNew(Template.GetName(), projectName, projectDirectory, Language.ToCliName(), customArgs: customNewArgs);
+        if (PreMadeSolution == null)
+        {
+            Directory.CreateDirectory(projectDirectory);
+            dotNetHelper.ExecuteNew(Template.GetName(), projectName, projectDirectory, Language.ToCliName(), customArgs: customNewArgs);
+        }
+        else
+        {
+            string PreMadeName;
+            if (PreMadeSolution.Contains(Path.DirectorySeparatorChar))
+            {
+                PreMadeName = PreMadeSolution.Split(Path.DirectorySeparatorChar).Last();
+            }
+            else
+            {
+                PreMadeName = PreMadeSolution;
+            }
+            projectDirectory = Path.Combine(testRoot, PreMadeName);
+            string fullPreMadePath = Path.Combine(Environment.CurrentDirectory, PreMadeSolution);
```

Comment:
Environment.CurrentDirectory is my guess to the culprit of the latest test failures. I was under the impression that it points to what is currently being run, (like the .dll that dotnet is running), but instead it's just where the console is before it starts running it.

Reply:
Depending on how the application is deployed (e.g. single file vs. normal) or run (dotnet vs. executable), finding resources alongside is not always super straightforward. I suggest System.AppContext.BaseDirectory https://learn.microsoft.com/en-us/dotnet/api/system.appcontext.basedirectory?view=net-7.0 as it should be able to find files alongside the app.

See https://stackoverflow.com/questions/74626616/finding-exe-dll-path-that-works-in-single-exe-and-running-through-dotnet-progr for more info.

Reply:
Yep AppContext.BaseDirectory does the trick. This is working now. Thanks!